nebula-logger/core/main/logger-engine/classes/LogEntryEventBuilder.cls

```diff
@@ -603,6 +603,50 @@ global with sharing class LogEntryEventBuilder {
         return this;
     }
 
+    /**
+     * @description Sets a field values on the builder's `LogEntryEvent__e` record
+     * @param  field      The `Schema.SObjectField` token of the field to populate
+     *                    on the builder's `LogEntryEvent__e` record
+     * @param  fieldValue The `Object` value to populate in the provided field
+     * @return The same instance of `LogEntryEventBuilder`, useful for chaining methods
+     */
+    global LogEntryEventBuilder setField(Schema.SObjectField field, Object fieldValue) {
```

Comment:
TODO decide if the method name should just be `set()` (or `put()`? 🤔 ) - the name would simpler & consistent for the overload/avoids the different names `setField()` and `setFields()`

Reply:
@jamessimone if you happen to have some time to spare, I'd love get your thoughts on the naming convention for these 2 new builder methods. Right now, I'm using:
- `setField(Schema.SObjectField field, Object fieldValue)` for setting a single field
- `setFields(Map<Schema.SObjectField, Object> fieldToValue)` for setting multiple fields

The slight difference in method name (`setField` vs `setFields`) could be fine, but I was also considering calling by methods `set()`, like:
- `set(Schema.SObjectField field, Object fieldValue)` for setting a single field
- `set(Map<Schema.SObjectField, Object> fieldToValue)` for setting multiple fields

I can't decide if I prefer shorter & consistent overload name `set()` - I think the current method names (`setField` and `setFields`) are a bit more descriptive, but I'm not always a fan of method names being singular/pluralized based on the parameters ("field" vs "fields").

Let me know what you think!

Reply:
I tend to prefer names that work for either argument, but `set` is part of the [reserved keywords](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_reserved_words.htm) though I think it's one of those weird ones where it's not technically enforced.

For a package - particularly in this day and age, without necessarily having intellisense on package methods - I could see the `setField` / `setFields` nomenclature helping.

Either way, my guess is whatever you choose will be fine!

Reply:
Good point about `set` being a reserved word - it does indeed work (I'm able to deploy a class with methods named `set()`), but it seems safer to avoid using that as a method name. I thought about using `put()` instead, but almost all of the existing methods in `LogEntryEventBuilder` use the convention `setSomething()` (e.g., `setMessage()`, `setRecord()`, `setHttpRequestDetails()`, etc.), so the use of the word `put` feels a little inconsistent, even though that's the SObject method being used.

I think I'm leaning towards staying with `setField()` and `setFields()`, but I'll think about it a bit more this week before wrapping up this PR.

Thanks for your help!

Reply:
I thought about this a bunch, and I've decided to go with `setField()` for both methods (making them overloads). In the end, the deciding factor for me was `setField()` (singular) is consistent with other builder method overloads, like `setRecord()`, which has overloads for `SObject`, `List<SObject>`, and `Map<Id, SObject>`.

I still struggle with which convention I prefer in general - but I love consistency, so `setField()` seems like the most consistent option relative to the rest of Nebula Logger's codebase.